src/pools/abstract-pool.ts

```diff
@@ -220,10 +220,11 @@ export abstract class AbstractPool<
    * @returns Worker.
    */
   protected chooseWorker (): Worker {
-    const chosenWorker = this.workers[this.nextWorkerIndex]
-    this.nextWorkerIndex++
-    this.nextWorkerIndex %= this.workers.length
-    return chosenWorker
+    this.nextWorkerIndex =
+      this.workers.length - 1 === this.nextWorkerIndex
+        ? 0
+        : this.nextWorkerIndex + 1
+    return this.workers[this.nextWorkerIndex]
```

Comment:
I think you can keep the nextWorkerIndex pre-calculation, it's a small optimisation but still.

Reply:
![image](https://user-images.githubusercontent.com/7195563/107863148-0a0e1200-6e52-11eb-9d2b-3e9010dca6c2.png)

It's even faster :rocket: